### PR TITLE
openpgp: disable check of short 2-byte hashes

### PR DESCRIPTION
**Pull request by maxtaco**
It was failing on @brentmaxwell's key. I can't think of any good this
check can do in terms of cryptography, so I think it's safe to disable it.
Disabling it does allow Brent's key to be processed successfully.

### CODE REVIEW

**Comment by maxtaco** [comment]
See this issue: https://github.com/keybase/client/issues/3905

File: openpgp/packet/public_key.go [hunk 1]
```diff
@@ -571,7 +571,11 @@ func (pk *PublicKey) VerifySignature(signed hash.Hash, sig *Signature) (err erro
 	hashBytes := signed.Sum(nil)
 
 	if hashBytes[0] != sig.HashTag[0] || hashBytes[1] != sig.HashTag[1] {
-		return errors.SignatureError("hash tag doesn't match")
+		// NOTE(maxtaco) 2016-08-22
+		// Don't do anything in this case. Some GPGs generate bad
+		// 2-byte hash prefixes, but GPG also doesn't seem to care on
+		// import. See BrentMaxwell's key. I think it's safe to disable
+		// this check!
```
Comment (oconnor663) [inline openpgp/packet/public_key.go]: Mild preference for just commenting out the whole thing in place, rather than leaving just part of the structure.

**Comment by oconnor663** [comment]
🚢 it. May our children forgive us.

**Pull request merged**